arangod/IResearch/IResearchViewCoordinator.cpp

```diff
@@ -340,11 +340,54 @@ bool IResearchViewCoordinator::emplace(
     uint64_t planVersion,
     LogicalView::PreCommitCallback const& preCommit
 ) {
+  auto& properties = info.isObject() ? info : emptyObjectSlice(); // if no 'info' then assume defaults
+  std::string error;
+
+  bool hasLinks = properties.hasKey("links");
+
   auto view = std::shared_ptr<IResearchViewCoordinator>(
     new IResearchViewCoordinator(vocbase, info, planVersion)
   );
-  auto& properties = info.isObject() ? info : emptyObjectSlice(); // if no 'info' then assume defaults
-  std::string error;
+
+  if (hasLinks) {
+    auto* engine = arangodb::ClusterInfo::instance();
+
+    if (!engine) {
+      return nullptr;
+    }
+
+    // check link auth as per https://github.com/arangodb/backlog/issues/459
+    if (arangodb::ExecContext::CURRENT) {
+      // check existing links
+      for (auto& entry: view->_collections) {
+        auto collection =
+          engine->getCollection(vocbase.name(), std::to_string(entry.first));
+
+        if (collection
+            && !arangodb::ExecContext::CURRENT->canUseCollection(vocbase.name(), collection->name(), arangodb::auth::Level::RO)) {
+          return nullptr;
```

Comment:
should link creation failure cause view creation failure?

Reply:
Should it say that everything is fine?